inc/fields/class-field-user-select.php

```diff
@@ -0,0 +1,174 @@
+		) );
+	}
+
+	/**
+	 * Output styles and templates used by user select field.
+	 */
+	public function action_shortcode_ui_loaded_editor() {
+		?>
+
+		<style>
+
+			.edit-shortcode-form .select2-container {
+				min-width: 300px;
+			}
+
+			.edit-shortcode-form .select2-container a {
+				transition: none;
+				-webkit-transition: none;
+			}
+
+			.wp-admin .select2-drop {
+				z-index: 160001;
+			}
+
+		</style>
+
+		<script type="text/html" id="tmpl-shortcode-ui-field-user-select">
+			<div class="field-block shortcode-ui-field-user-select shortcode-ui-attribute-{{ data.attr }}">
+				<label for="{{ data.id }}">{{{ data.label }}}</label>
+				<input type="text" name="{{ data.attr }}" id="{{ data.id }}" value="{{ data.value }}" class="shortcode-ui-user-select" />
+				<# if ( typeof data.description == 'string' ) { #>
+					<p class="description">{{{ data.description }}}</p>
+				<# } #>
+			</div>
+		</script>
+
+		<?php
+	}
+
+	/**
+	 * Ajax handler for select2 user field queries.
+	 * Output JSON containing user data.
+	 * Requires that shortcode, attr and nonce are passed.
+	 * Requires that the field has been correctly registered and can be found in $this->post_fields
+	 * Supports passing page number and search query string.
+	 *
+	 * @return null
+	 */
+	public function action_wp_ajax_shortcode_ui_user_field() {
+
+		$nonce               = isset( $_GET['nonce'] ) ? sanitize_text_field( $_GET['nonce'] ) : null;
+		$requested_shortcode = isset( $_GET['shortcode'] ) ? sanitize_text_field( $_GET['shortcode'] ) : null;
+		$requested_attr      = isset( $_GET['attr'] ) ? sanitize_text_field( $_GET['attr'] ) : null;
+		$response            = array( 'users' => array(), 'found_users' => 0, 'users_per_page' => 0 );
+
+		if ( ! wp_verify_nonce( $nonce, 'shortcode_ui_field_user_select' ) ) {
+			wp_send_json_error( $response );
+		}
+
+		$shortcodes = Shortcode_UI::get_instance()->get_shortcodes();
+
+		// Shortcode not found.
+		if ( ! isset( $shortcodes[ $requested_shortcode ] ) ) {
+			wp_send_json_error( $response );
+		}
+
+		$shortcode = $shortcodes[ $requested_shortcode ];
+
+		// Defaults user query args.
+		$query_args['search_columns'] = array( 'ID', 'user_login', 'user_nicename', 'user_email' );
+		$query_args['number']         = 10;
+
+		// Include selected users.
+		if ( isset( $_GET['include'] ) ) {
+			$query_args['include'] = is_array( $_GET['include'] ) ? $_GET['include'] : explode( ',', $_GET['include'] );
+			$query_args['include'] = array_map( 'absint', $query_args['include'] );
+		}
+
+		// Supports WP_User_Query query args.
+		foreach ( $shortcode['attrs'] as $attr ) {
+			if ( $attr['attr'] === $requested_attr && isset( $attr['query'] ) ) {
+				$query_args = $attr['query'];
+			}
+		}
+
+		if ( isset( $_GET['page'] ) ) {
+			$query_args['paged'] = sanitize_text_field( $_GET['page'] );
+		}
+
+		if ( ! empty( $_GET['s'] ) ) {
+			$query_args['search'] = '*' . sanitize_text_field( $_GET['s'] ) . '*';
```

Comment:
@westonruter in the second code chunk you mean `if ( ! empty( $s ) ) {` as it can be non-empty before sanitisation and empty after?